Push contracts and seed to NPM

jksf opened the pull request:
This PR depends on #32. Do not merge before.

# Usage examples

### `package.json`
```json
  "scripts": {
    "seed": "signhash-seed",
    "seed-explore": "npm explore signhash-contracts -- npm run seed:js"
  },
  "dependencies": {
    "signhash-contracts": "^0.2.0"
  }
```

`seed` and `seed-explore` do exactly the same

### `example.js`
```javascript
const signhash = require('signhash-contracts');
console.log(signhash.SignHashContract.abi);
```

Commit:
Create Seed script

Compiles contracts, starts TestRPC and runs migrations.

Commit:
Configure NPM package

Commit:
Prepare assets before publishing

Commit:
Signal process exit code on exception

Commit:
Update compiled contracts

Commit:
Export compiled contract

Commit:
Bump version number to v0.2.0

biern commented on package.json:
```diff
@@ -45,20 +55,30 @@
+    "truffle": "^4.0.1",
     "tslint": "^5.7.0",
     "typescript": "^2.5.2"
   }
```
So, unfortunately, removing the `migrate` and moving `truffle` to `devDependencies` breaks the webapp build pipeline. Webapp relies on being able to run `contracts:migrate` as a single command. When truffle is in dev dependencies it's not installed when installing `signhash-contracts` as a dependency and thus it is impossible to run migrations via CLI.

Commit:
Restore compatibility with webapp

Restore `migrate` command.
Move Truffle from dev to main dependencies.

Commit:
Bump version number to 0.2.1

jksf commented:
Fixed compatibility issues with webapp in #32. I rebased this PR to include changes from #32 and published new NPM package (`0.2.1`). Ready to merge right after #32.
